src/Lean/Widget/InteractiveGoal.lean

```diff
@@ -33,6 +45,12 @@ structure InteractiveGoal where
   name of the MVar that it is a goal for.)
   This is none when we are showing a term goal. -/
   mvarId? : Option MVarId := none
+  /-- If true, the goal was not present on the previous tactic state. -/
+  isInserted?: Option Bool := none
+  /-- If true, the goal will be removed on the next tactic state. -/
+  isRemoved? : Option Bool := none
+  /-- An optional message string to show the user after the goal. This is useful for debugging.-/
```

Comment:
If the `message?` fields are only used for development, do we need to keep them around? What's the advantage of sending back a `message` rather than throwing an exception (which would also show up in the infoview)?

Reply:
They are just for development. It's harder to debug if I used exceptions because I wouldn't be able to look at the tactic state.

Reply:
I will remove at the end

Reply:
removed